drivers/at86rf2xx/Makefile.dep

```diff
@@ -1,11 +1,14 @@
 DEFAULT_MODULE += auto_init_at86rf2xx
-DEFAULT_MODULE += netdev_ieee802154_oqpsk
 
 USEMODULE += ztimer
 USEMODULE += ztimer_usec
-USEMODULE += ieee802154
-USEMODULE += netdev_ieee802154
-USEMODULE += netdev_legacy_api
+USEMODULE += event_thread
+USEMODULE += event_thread_highest
+
+ifneq (,$(filter netdev,$(USEMODULE)))
+  USEMODULE += netdev_ieee802154_submac
+  DEFAULT_MODULE += netdev_ieee802154_oqpsk
+endif
 
 # only needed for SPI based variants
 ifeq (,$(filter at86rfa1 at86rfr2,$(USEMODULE)))
```

Comment:
```suggestion
ifeq (,$(filter at86rfa1 at86rfr2,$(USEMODULE)))
  ifneq (,$(filter ieee802154_security,$(USEMODULE)))
    USEMODULE += at86rf2xx_aes_spi
  endif
```

Reply:
I would keep it optional whether to use hardware offloading. IIRC pings were actually faster without.

Reply:
That's probably due to all the `spi_aquire()` - we should move those to the entry points of rf_ops instead of doing it for every register read, but that's for a follow-up.

Reply:
and `at86rf230` does not have this feature

Reply:
should I then just leave it as it is? or catch the `at86rf230` case?

Reply:
I would treat it as a nice to have feature and not enable it in this PR. @benpicco?

Reply:
Under what condition would you enable the feature then?
I mean it *should* be able to reduce the binary size because we don't need the software implementation then.

I thought it would also safe energy / be faster, but I didn't test it.

Reply:
Ok, we would not need `cipher_modes` if we know there is only `at86rf2xx` in use and for example not an additional `at86rf215` or other 802.15.4 transceiver (I know its probably an uncommon case). To gain more from it we should test if we need `cipher_ops` or `at86rf2xx_aes_spi`.

```
USEMODULE += ieee802154_security

   text    data     bss     dec     hex filename
 106496     272   22328  129096   1f848 /home/fabian/forks/RIOT/examples/gnrc_networking/bin/miot-nucleo-f767zi/gnrc_networking.elf

```

```
ping -s 120 -c 50 fe80::4c49:6f54:4f76:102%7

2023-02-24 19:41:44,944 # --- fe80::4c49:6f54:4f76:102%7 PING statistics ---
2023-02-24 19:41:44,950 # 50 packets transmitted, 50 packets received, 0% packet loss
2023-02-24 19:41:44,953 # round-trip min/avg/max = 22.881/24.924/28.171 ms
```

```
USEMODULE += "ieee802154_security at86rf2xx_aes_spi"

   text    data     bss     dec     hex filename
 107328     272   22328  129928   1fb88 /home/fabian/forks/RIOT/examples/gnrc_networking/bin/miot-nucleo-f767zi/gnrc_networking.elf

```

```
ping -s 120 -c 50 fe80::4c49:6f54:4f76:102%7

2023-02-24 19:45:36,758 # --- fe80::4c49:6f54:4f76:102%7 PING statistics ---
2023-02-24 19:45:36,764 # 50 packets transmitted, 50 packets received, 0% packet loss
2023-02-24 19:45:36,768 # round-trip min/avg/max = 26.763/29.079/31.253 ms
```

Reply:
We should probably introduce a `netdev_ieee802154_submac_soft_cipher` (analogous to the proposed `netdev_ieee802154_submac_soft_retrans`) that drivers without HW crypto can then select.

Reply:
Sounds good. If we have that the right modules are selected dynamically base on the transceivers.
But changing  the dependency  of `at86rf2xx_aes_spi` in this PR is not really advisable.
Or would you still do it here?

Reply:
I have no strong feelings about this, I was just surprised that it was not build in.
That's probably also why @jia200x didn't notice that the current version crashes when `at86rf2xx_aes_spi` is enabled, because when you want to test `ieee802154_security` it just works - because the HW acceleration is not used.

Reply:
then should I just leave it as it is for now? We can address that in a follow up.

Reply:
Please fix the crash, it's a very simple change